File: src/vendors/wc3maptranslator/data/Camera.ts
import { type angle } from '../CommonInterfaces';

interface Camera {
  target: CameraTarget
  offsetZ: number
  rotation: angle
  aoa: angle // angle of attack
  distance: number
  roll: number
  fov: angle // field of view
  farClipping: number
  nearClipping: number
  localPitch: number
  localYaw: number
  localRoll: number
  name: string
}

interface CameraTarget {
  x: number
  y: number
}

export type { Camera, CameraTarget };

File: golang/internal/wc3/data/camera.go
package data

import "github.com/pqhuy98/wow-converter/internal/wc3"

// CameraTarget is the XY target of a camera.
type CameraTarget struct {
	X float32
	Y float32
}

// Camera is a war3map.w3c camera definition.
type Camera struct {
	Target       CameraTarget
	OffsetZ      float32
	Rotation     wc3.Angle
	Aoa          wc3.Angle
	Distance     float32
	Roll         float32
	Fov          wc3.Angle
	FarClipping  float32
	NearClipping float32
	LocalPitch   float32
	LocalYaw     float32
	LocalRoll    float32
	Name         string
}
